### DISCUSSION TITLE
Why and how is diagnostic data collected and sent back to the MinIO organization

### DISCUSSION THREAD

**Discussion by rickpeters** [minio/minio#12826] [Q&A] [answered]
Hello,

I do not know if this is the right place to ask but I have not been able to find information in other places, so I will try anyway.

We got an email from MinIO marketing asking us if we would like to discuss the possibility of procuring a commercial license for our use of MinIO. In this mail they were able to show us numbers on the amount of usage of Minio in the organization and were also able to name specific MinIO hostnames in our (private cloud) environment.
Until now I did not know that usage statistics are gathered and communicated back towards MinIO and I think that at the very least there should be some kind of confirmation before software is allowed to do so.

**Does anybody know where these statistics are gathered, how they are sent back to minIO and if it is possble to disable this behaviour?**

To be clear, in itself I am not opposed to having a license model and not opposed to paying if necessary. Until now I didn't see the need to think about the AGPL v3 consequences because I think for us there are none (we only install the binary and expose the interface internally for object-storage and we do not redistribite anything).

However I am opposed towards gathering usage statistics about internal details of our environment without us knowing about it.

regards,
Rick

**Comment by harshavardhana** [comment, marked as answer]
We do not collect any information regarding your systems,  all the software does is send a ping to https://dl.min.io when you restart or start the server to check for  latest releases to remind you to upgrade.

If you don't want this check you can always set MINIO_UPDATE=off 

The information that is available in the update check is used to understand that usage patterns that's all.

https://github.com/minio/minio/blob/master/cmd/update.go

The code on what it does.

**Reply by rickpeters** [reply]
Hi,
thanks for the answer. As I see it the usage pattern would only show that a lot of restarts are necessary or being performed, so not really a good statistic on usage. Most of our systems do not have an internet connection configured so this ping wouldn't work.

Anyway, I understand now what happens and it is totally reasonable that this is triggered en tries to do a check whether a new version is available. Nothing specific is gathered, compliments on that 👍

regards,
Rick
